core/src/test/kotlin/io/specmatic/core/CalculatePathTest.kt

```diff
@@ -726,4 +726,205 @@ internal class CalculatePathTest {
         
         assertThat(paths).containsExactlyInAnyOrder("{Person}.officeAddress{AddressRef}", "{Person}.homeAddress{Address}")
     }
+    
+    @Test
+    fun `calculatePath should handle deeply nested AnyPattern with typeAlias`() {
+        // Test case 1: Nested structure with typeAlias at multiple levels
+        val level3Pattern = JSONObjectPattern(
+            pattern = mapOf(
+                "data" to AnyPattern(listOf(StringPattern(), NumberPattern()))
+            ),
+            typeAlias = "Level3Object"
+        )
+        
+        val level2Pattern = JSONObjectPattern(
+            pattern = mapOf(
+                "level3" to level3Pattern
+            ),
+            typeAlias = "Level2Object"
+        )
+        
+        val level1Pattern = JSONObjectPattern(
+            pattern = mapOf(
+                "level2" to level2Pattern
+            ),
+            typeAlias = "Level1Object"
+        )
+
+        val value = JSONObjectValue(mapOf(
+            "level2" to JSONObjectValue(mapOf(
+                "level3" to JSONObjectValue(mapOf(
+                    "data" to StringValue("test")
+                ))
+            ))
+        ))
+
+        val paths = level1Pattern.calculatePath(value, Resolver())
+        
+        assertThat(paths).containsExactly("{Level1Object}.level2.{Level2Object}.level3.{Level3Object}.data{string}")
```

Comment:
This path set should have contained: "{Level1Object}.level2{Level2Object}.level3{Level3Object}.data{string}".

Update the test accordingly, and then update the application code to pass the test.